packages/grafana-ui/src/components/FileUpload/FileUpload.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { FC, FormEvent, useState } from 'react';
+import { GrafanaTheme } from '@grafana/data';
+import { css, cx } from 'emotion';
+import { getFormStyles, Icon } from '../index';
+import { stylesFactory, useTheme } from '../../themes';
+
+export interface Props {
+  onFileUpload: (event: FormEvent<HTMLInputElement>) => void;
+  /** Custom upload label text */
+  label?: string;
+  /** Accepted file extensions */
+  accept?: string;
+  className?: string;
+}
+
+function trimFileName(fileName: string) {
+  const nameLength = 16;
+  const delimiter = fileName.lastIndexOf('.');
+  const extension = fileName.substring(delimiter);
+  const file = fileName.substring(0, delimiter);
+
+  if (file.length < nameLength) {
+    return file;
+  }
+
+  return file.substring(0, nameLength) + '...' + extension;
+}
+
+export const FileUpload: FC<Props> = ({ onFileUpload, className, label = 'Upload file', accept = '*' }) => {
```

Comment:
I wonder if this shouldn't belong to forms namespace maybe? Thoughts @tskarhed ?